src/query/createCompoundQuery.ts

```diff
@@ -33,36 +49,56 @@ const createCompoundQuery: QueryFactory = compose<CompoundQuery<{}>, QueryOption
 		}, data));
 	},
 
-	withQuery(this: Query<{}>, query: Query<{}>): CompoundQuery<{}> {
+	withQuery(this: CompoundQuery<{}>, query: Query<{}>): CompoundQuery<{}> {
```

Comment:
Sorry if this is completely wrong but could `withQuery<T>(this: CompoundQuery<T>, query: Query<T>): CompoundQuery<T> {` work? I see we use `{}` all the time in stores to satisfy generics and wonder if there is any point in having the generics if we always just internally specify `{}` or `any` as the generic.